test/widgets/emoji_reaction_test.dart

```diff
@@ -466,6 +464,87 @@ void main() {
 
       debugNetworkImageHttpClientProvider = null;
     });
+
+    group('handle view paddings', () {
+      const screenHeight = 400.0;
+
+      late double scrollViewTopEdgeOffset;
+      late double scrollViewBottomEdgeOffset;
+      final scrollViewFinder = findInPicker(find.bySubtype<ScrollView>());
+
+      final listEntryFinder = find.byType(EmojiPickerListEntry);
+      double getListEntriesTopEdgeOffset(WidgetTester tester) =>
+        tester.getTopLeft(listEntryFinder.first).dy;
+      double getListEntriesBottomEdgeOffset(WidgetTester tester) =>
+        tester.getBottomLeft(listEntryFinder.last).dy;
+
+      Future<void> prepare(WidgetTester tester, {
+        required FakeViewPadding viewPadding,
+      }) async {
+        addTearDown(tester.view.reset);
+        tester.view.physicalSize = Size(640, screenHeight);
+        // This makes it easier to convert between device pixels used for
+        // [FakeViewPadding] and logical pixels used in tests.
+        tester.view.devicePixelRatio = 1.0;
+
+        tester.view.viewPadding = viewPadding;
+        tester.view.padding = viewPadding;
+
+        final message = eg.streamMessage();
+        await setupEmojiPicker(tester,
+          message: message, narrow: TopicNarrow.ofMessage(message));
+
+        scrollViewTopEdgeOffset = tester.getTopLeft(scrollViewFinder).dy;
+        scrollViewBottomEdgeOffset = tester.getBottomLeft(scrollViewFinder).dy;
+        final scrollViewHeight = tester.getSize(scrollViewFinder).height;
```

Comment:
nit: These facts can all be bundled together as a Rect:
```suggestion
        scrollViewRect = tester.getRect(scrollViewFinder);
```

Then references below to these three variables can instead use `.top`, `.bottom`, `.height` on the Rect.

Reply:
Neat! I think we can make a helper for the list entries that returns a `Rect` too:

```dart
      Rect getListEntriesRect(WidgetTester tester) =>
        tester.getRect(find.byType(EmojiPickerListEntry).first)
          .expandToInclude(tester.getRect(find.byType(EmojiPickerListEntry).last));
```